lib/expression/callbacks/standard.ex

```diff
@@ -1488,10 +1488,12 @@ defmodule Expression.Callbacks.Standard do
   """
   @expression_doc expression: "has_pattern(\"Buy cheese please\", \"buy (\\w+)\")", result: true
   @expression_doc expression: "has_pattern(\"Sell cheese please\", \"buy (\\w+)\")", result: false
+  @expression_doc expression: "has_pattern(nil, \"buy (\\w+)\")", result: false
   def has_pattern(ctx, expression, pattern) do
     [expression, pattern] = eval_args!([expression, pattern], ctx)
 
-    with {:ok, regex} <- Regex.compile(String.trim(pattern), "i"),
+    with false <- is_nil(expression),
```

Comment:
@davidkhk following the [Good and bad Elixir](https://keathley.io/blog/good-and-bad-elixir.html) reference, we can maybe change this a bit to "State what you want, not what you don’t"

```suggestion
    with true <- is_binary(expression),
```